kvmagent/kvmagent/plugins/storage_device.py

```
@@ -1408,17 +1446,43 @@ def connect_nvme_controller(self, cmd):
         for nqn in discovered_nqns:
             r, o, e = bash.bash_roe("timeout 60 nvme connect -a %s -s %s -t %s --nqn %s" % (cmd.ip, cmd.port, cmd.transport, nqn))
-            for controller in self.get_nvme_subsystem_controllers(nqn):
-                if controller.transport == cmd.transport and controller.address == "traddr=%s,trsvcid=%s" % (cmd.ip, cmd.port):
-                    any_nqn_connected = True
-                    wwids = wwids.union(controller.wwids)
-                    break
-            if not any_nqn_connected:
+
+            def matching_controllers():
+                matched = []
+                for controller in self.get_nvme_subsystem_controllers(nqn):
```

Ruff (0.15.15)

[warning] Function definition does not bind loop variable `nqn`

(B023)